algorithms/src/std_algorithms/impl/Kokkos_Constraints.hpp

```diff
@@ -4,9 +4,18 @@
 #ifndef KOKKOS_STD_ALGORITHMS_CONSTRAINTS_HPP_
 #define KOKKOS_STD_ALGORITHMS_CONSTRAINTS_HPP_
 
+#include <Kokkos_Macros.hpp>
+#ifdef KOKKOS_ENABLE_EXPERIMENTAL_CXX20_MODULES
+import kokkos.core;
+#else
 #include <Kokkos_Core.hpp>
+#endif
+#include <Kokkos_Assert.hpp>
```

Comment:
What will we want to do for the few standalone headers we have (Timer, TypeInfo, etc.)? It would make sense to me to include them in the core module or have a separate `utils` module ...

Reply:
`TypeInfo` isn't public. `Timer` is included in the module. For now, everything should be part of the `kokkos.core` module. I'm not sure if we want to factor more modules out later (and decide if `kokkos.core` should depend on it. 
For now, the module doesn't include `desul` symbols but `mdspan` symbols (since the latter is in the `Kokkos` namespace). I would expect `mdspan` to have its own module in the end and we can then decide if it should be included or not.